hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -200,18 +201,10 @@ private void adjustCryptoAllowance(BalanceChange change, AccountID ownerID) {
 	private void adjustNftAllowance(final BalanceChange change, final AccountID ownerID) {
 		final var allowanceId = FcTokenAllowanceId.from(
 				change.getToken().asEntityNum(), EntityNum.fromAccountId(change.getPayerID()));
-		final var nftAllowances = new TreeMap<>(
-				(Map<FcTokenAllowanceId, FcTokenAllowance>) accountsLedger.get(ownerID, NFT_ALLOWANCES));
-		final var currentAllowance = nftAllowances.get(allowanceId);
-		if (!currentAllowance.isApprovedForAll()) {
-			var mutableAllowanceList = new ArrayList<>(currentAllowance.getSerialNumbers());
-			mutableAllowanceList.remove(change.serialNo());
-			if (mutableAllowanceList.isEmpty()) {
-				nftAllowances.remove(allowanceId);
-			} else {
-				nftAllowances.put(allowanceId, FcTokenAllowance.from(mutableAllowanceList));
-			}
-			accountsLedger.set(ownerID, NFT_ALLOWANCES, nftAllowances);
+		final var approveForAllNftsAllowances = (Set<FcTokenAllowanceId>) accountsLedger.get(ownerID, APPROVE_FOR_ALL_NFTS_ALLOWANCES);
+
+		if (!approveForAllNftsAllowances.contains(allowanceId)) {
```

Comment:
Even if `approveForAll` exists , if the spenderId is on nft , we should clear it right ? Since both are treated separately . There can be a case where we have `approveForAll` and explicit allowance given

Reply:
Good point. Made the changes.